src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -264,6 +264,17 @@ void primitive_inst::update_shape() {
         }
     }
 
+    if (get_node().is_type<read_value>()) {
+        const auto& variable_id = get_node().as<read_value>().get_primitive()->variable_id;
+        auto new_layout = get_network().get_variable(variable_id).get_layout();
+        if (!_impl_params->state_layout.has_value() || _impl_params->state_layout.value() != new_layout) {
+            _impl_params->state_layout = new_layout;
+            input_shape_changed = true;
+        }
+
+        input_shape_changed = true;
```

Comment:
Hmm, do we need always force `input_shape_changed = true;` here?

Reply:
Removed